avrohugger-core/src/main/scala/Generator.scala

```diff
@@ -41,58 +42,65 @@ class Generator(format: SourceFormat,
   //////////////// methods for writing definitions out to file /////////////////
   def schemaToFile(
     schema: Schema,
-    outDir: String = defaultOutputDir): Unit = {
+    outDir: String = defaultOutputDir,
+    restrictedFields: Boolean = false): Unit = {
```

Comment:
Looking great mostly. I assume this was meant to be like you did the other methods, with the resitrictedFields commented out? In this case I'm in favor of using the global restrictedFieldNumber in order to keep the api slim and unchanging across versions if we can help it.

And while these methods are covered in various tests, the above inconsistency actually results in a bug where the restrictedFields set in the Generator can be ignored, suggesting tests are needed here too: I think a large (>22) Schema and a large IDL, each in String format and in a File, and each of those toString and toFile (1 avsc and 1 avdl reused in 8 total examples) would help. I'll fix the odd method, remove the commented params, and publish after the tests are added, this weekend if you don't beat me to it.